rules/windows/privilege_escalation_reg_service_imagepath_mod.toml

```diff
@@ -0,0 +1,202 @@
+[metadata]
+creation_date = "2024/06/05"
+integration = ["endpoint", "windows"]
+maturity = "production"
+updated_date = "2024/06/05"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies registry modifications to default services that could enable privilege escalation to SYSTEM. Attackers with
+privileges from groups like Server Operators may change the ImagePath of services to executables under their control or
+to execute commands.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.registry-*", "logs-windows.sysmon_operational-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Privilege Escalation via Service ImagePath Modification"
+references = [
+    "https://cube0x0.github.io/Pocing-Beyond-DA/"
+]
+risk_score = 47
+rule_id = "b66b7e2b-d50a-49b9-a6fc-3a383baedc6b"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Execution",
+    "Tactic: Privilege Escalation",
+    "Data Source: Elastic Defend",
+    "Data Source: Sysmon"
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+registry where host.os.type == "windows" and event.type == "change" and
+  event.action == "modification" and registry.value == "ImagePath" and
+  registry.key : (
```

Comment:
for better perf use `endswith~(registry.key, "ADWS", "AppHostSvc" ...)`

Reply:
Isn't that the same as using the wildcard? I think this was the conclusion

Reply:
Conceptually in query languages, since `SYSTEM\\*ControlSet*\\Services` is the same amongst all these, it is best practice to use the `endsWith()` function and not abuse wildcards. Right now this will do multiple multiple term searches, scanning each character in the potential matches and comparing substrings at different offsets. Whereas the  `endsWith()` function should check the suffix of the string only, which is a simpler operation. 

Often the performance issues we run into are memory related. Wildcard usage needs to maintain multiple states of the string to evaluate different parts of the potential pattern, whereas optimized functions will operate on a fixed-length substring.

If you were replacing multiple wildcard terms with multiple language functions, the trade-off may not differ much, but since they are have the same prefix, we should use the function instead.

Reply:
I mean more like, it is basically the same if I use `endswith~(registry.key, "ADWS")` or `registry.key : "*ADWS"`, right?

Reply:
At the moment this is all conceptual. At a high-level, yes they are similar in what is being achieved. At a lower level they do differ and my assumption is that the functions are a better practice than wildcards in this instance based on what I shared -> https://github.com/elastic/detection-rules/pull/3757#discussion_r1634779550